Sources/ComposableArchitectureMacros/ViewActionMacro.swift

```diff
@@ -0,0 +1,169 @@
+@preconcurrency import SwiftDiagnostics
+import SwiftSyntax
+import SwiftSyntaxMacroExpansion
+import SwiftSyntaxMacros
+
+public struct ViewActionMacro: MemberMacro {
+  public static func expansion<Declaration, Context>(
+    of node: SwiftSyntax.AttributeSyntax,
+    providingMembersOf declaration: Declaration,
+    in context: Context
+  ) throws -> [SwiftSyntax.DeclSyntax]
+  where Declaration: SwiftSyntax.DeclGroupSyntax, Context: SwiftSyntaxMacros.MacroExpansionContext {
+    guard declaration.hasStoreVariable
+    else {
+      // TODO: Fix it to add `let store: StoreOf<<#Feature#>>`?
+      context.diagnose(
+        Diagnostic(
+          node: declaration,
+          message: MacroExpansionErrorMessage(
+            """
+            @ViewAction macro requires \
+            \(declaration.identifierDescription.map { "'\($0)' " } ?? "") to have a 'store' \
+            property of type 'Store'.
+            """
+          ),
+          fixIt: .replace(
+            message: MacroExpansionFixItMessage("Add 'let store'"),
+            oldNode: declaration,
+            newNode: DeclReferenceExprSyntax(baseName: "send")
+          )
+        )
+      )
+      return []
+    }
+
+    declaration.diagnoseDirectStoreDotSend(
+      declaration: declaration,
+      context: context
+    )
+
+    guard
+      case let .argumentList(arguments) = node.arguments,
+      arguments.count == 1
+    else { return [] }
+    guard
+      let memberAccessExpr = arguments.first?.expression.as(MemberAccessExprSyntax.self)
+    else { return [] }
+    let rawType = String("\(memberAccessExpr)".dropLast(5))
+
+    return [
+      """
+      @discardableResult
+      fileprivate func send(_ action: \(raw: rawType).Action.View) -> StoreTask {
+        store.send(.view(action))
+      }
+      @discardableResult
+      fileprivate func send(_ action: \(raw: rawType).Action.View, animation: Animation?) -> StoreTask {
+        store.send(.view(action), animation: animation)
+      }
+      @discardableResult
+      fileprivate func send(_ action: \(raw: rawType).Action.View, transaction: Transaction) -> StoreTask {
+        store.send(.view(action), transaction: transaction)
+      }
```

Comment:
Adding the 3 main variations of `send` for now. Can reconsider `send(_:while:)` later.